mungegithub/mungers/submit-queue_test.go

```diff
@@ -902,6 +904,8 @@ func TestSubmitQueue(t *testing.T) {
 				// so we don't want to modify the PR there. Instead we need
 				// to wait until the second time we check Jenkins, which happens
 				// we did the IsMerged() check.
+				testChecksLock.Lock()
+				defer testChecksLock.Unlock()
```

Comment:
I think defers happen when the function returns, not when the loop continues. Is this what you intend?